WordPress/gutenberg
# Style engine: use style engine for block supports CSS in editor

## What?

- Using the style engine to return gradient values and font family. _Note_: font families are mapped from theme.json presets so the style engine won't return any values unless they are custom.
- Now that the style engine supports block supports properties (except duotone), removing the __EXPERIMENTAL_STYLE_PROPERTY loop in styles hook.
- Updated tests

## Why?
As far as block supports styles in the editor are concerned, the JS style engine is complete.

The JS style engine parses and compiles block supports CSS for the block and site editors.

The notable omissions are:

1. Elements, which is handled separately in `withElementsStyles()`
2. Duotone, also handled separately by `withDuotoneStyles()`
3. Block gap. This is handled by `withLayoutStyles()`

In this respect, the JS package behaves similarly to the PHP version.

## Testing Instructions

We are testing all possible variations of a block's `attributes.style` object.

In the block editor, add several blocks - group, any text block, and image/gallery for example - whose block supports cover the full suite of available style properties (`spacing`, `typography`, `color`, `border` ).

Apply custom values where possible.

Ensure that your changes appear as expected in the block editor and in the frontend.

**Repeat for the site editor.**

<details>

<summary>Here is some example block code to play with.</summary>

```html

<div class="wp-block-group has-background" style="background:linear-gradient(135deg,rgb(255,203,112) 0%,rgb(33,32,33) 42%,rgb(65,88,208) 100%);margin-top:74px;margin-bottom:74px;padding-top:62px;padding-right:62px;padding-bottom:62px;padding-left:62px">
<p class="has-text-color has-background" style="color:#0d0e0e;background-color:#a626424d;font-size:44px;font-style:normal;font-weight:600;letter-spacing:14px;line-height:2.4;text-transform:uppercase">Group block with gradient background</p>
</div>



<figure class="wp-block-image size-full has-custom-border"><img src="http://localhost:8888/wp-content/uploads/2022/07/puppy.jpg" alt="puppy-alt" class="wp-image-43" style="border-top-left-radius:49px;border-top-right-radius:49px;border-bottom-left-radius:49px;border-bottom-right-radius:0px;border-top-width:46px;border-right-color:#de2727;border-right-width:79px;border-bottom-color:#031019;border-bottom-width:29px;border-left-color:#3bf1ae;border-left-style:dotted;border-left-width:30px"/><figcaption class="wp-element-caption">puppy</figcaption></figure>



<h2>Heading</h2>

```

</details>
